### DISCUSSION TITLE
Parse Time Limit exceeded

### DISCUSSION THREAD

**Discussion by cwichoski** [CordyJ/OpenTxl#42] [Q&A] [answered]
Hi

Using TXL for first time, I used obsolete version of FreeTXL v10.8b, I can see a message about Parse time limit like this:

`[/tmp/sample.pas, pas.txl] : TXL0127E input file '/tmp/sample.pas' - Parse time limit (26600000 cycles) exceeded`

bellow of this it's show a false positive about Syntax error, but it's because the parser was interrupted by parse time limit. (I tested with same syntax with a small source code and syntax is ok)

Then I found OpenTXL v11.3.5, and give it a try, but in this it only show the false positive error same as FreeTXL.

I tried to find how to increase this Parse time limit but dont find any documentation about. Exists any parameter to increase this limit?

**Comment by cwichoski** [comment]
After review the TXL grammar I found a mistake in grammar, removed that and not getting Systax error anymore. Then in truth with FreeTXL I had two problems, then OpenTXL solved the problem with parse time limit.

**Comment by CordyJ** [comment, marked as answer]
txl -s 1000 will increase the time limit.

But usually hitting the parse time limit means a syntax error in your input file,
either because the grammar does not match something or because the input its malformed.

JC


On Aug 3, 2024, at 7:50 PM, Clóvis Wichoski ***@***.***> wrote:



Hi

Using TXL for first time, I used obsolete version of FreeTXL v10.8b, I can see a message about Parse time limit like this:

[/tmp/sample.pas, pas.txl] : TXL0127E input file '/tmp/sample.pas' - Parse time limit (26600000 cycles) exceeded

bellow of this it's show a false positive about Syntax error, but it's because the parser was interrupted by parse time limit. (I tested with same syntax with a small source code and syntax is ok)

Then I found OpenTXL v11.3.5, and give it a try, but in this it only show the false positive error same as FreeTXL.

I tried to find how to increase this Parse time limit but dont find any documentation about. Exists any parameter to increase this limit?
